cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -236,12 +237,28 @@ func (cm *certManagerClient) PlanUpgrade(ctx context.Context) (CertManagerUpgrad
 	}, nil
 }
 
+// getFilteredCertManagerResources gets all relevant objects for a cert-manager deployment
+// it also includes objects in the kube-system namespace because leader election namespace
+// is hard coded to kube-system https://github.com/cert-manager/cert-manager/issues/6716
+// it also deletes components that are created by Kubernetes and won't be used to see if
+// new objects need to be created for a cert-manager upgrade.
```

Comment:
nit: some wordsmithing:

```suggestion
// getFilteredCertManagerResources gets all relevant objects for a cert-manager installation.
// It also includes relevant resources in the kube-system namespace, which is used by cert-manager
// for leader election (https://github.com/cert-manager/cert-manager/issues/6716).
// It excludes resources that are related to the cert-manager installation, but not relevant
// to the cert-manager upgrade.
```

Reply:
words were smith'ed thanks for the feedback